ERCS/erc-7832.md

```diff
@@ -0,0 +1,355 @@
+
+    #### `lastUpdateTimestamp`
+    **Type**: `uint256`  
+    **Description**:  
+    Timestamp of the last time the contract terms were updated (e.g., minting fees and creator signature fees). It is used to determine when the contract's terms can be updated again.
+
+    #### `UPDATE_INTERVAL`
+    **Type**: `uint256` (constant)  
+    **Description**:  
+    The time interval between contract terms updates. **MUST** be fixed to <ins>30 days</ins>.
+
+    #### `CREATOR_ROLE`
+    **Type**: `bytes32` (constant)  
+    **Description**:  
+    The role identifier for creators in the system. 
+
+    #### `CONTRIBUTOR_ROLE`
+    **Type**: `bytes32` (constant)  
+    **Description**:  
+    The role identifier for contributors in the system. 
+
+    #### `mintsPerUserInCycle`
+    **Type**: `mapping(address => uint256)`  
+    **Description**:  
+    A mapping that tracks the number of mints a user has performed in the current cycle. It is used to enforce the maximum minting limit per user.
+
+    ---
+
+2. **Events**
+    #### `CreatorTermsUpdated(uint256 mintBaseFee, uint256 creatorSignatureFee, uint256 maxMintsPerUserInCycle)`
+    **Description**:  
+    Emitted when the contract terms related to minting are updated by the `DEFAULT_ADMIN_ROLE`.
+
+
+    #### `DonationReceived(address from, address to, uint256 amount)`
+    **Description**:  
+    Emitted when a user donates ETH to a creator. This event tracks the details of the donation, including the donor's address, the recipient's address, and the donation amount.
+
+    **Parameters**:  
+    - `from`: The address of the user making the donation.  
+    - `to`: The address of the creator receiving the donation.  
+    - `amount`: The amount of ETH donated.
+
+
+    ---
+
+3. **Modifiers**
+    #### `onlyIfNotPaused`
+    **Description**:  
+    Ensures that the contract **MUST NOT** be paused when executing a function.
+
+    ---
+
+4. **Functions**
+    #### `function safeMint(string memory uri)`
+    **Type**: Public and `payable` function  
+    **Description**:  
+    Allows the caller to mint a new token to their address with a provided URI.
+
+    ```solidity
+    function safeMint(string memory uri)
+    public
+    payable
+    override 
+    onlyIfNotPaused
+    nonReentrant
+    onlyRole(CREATOR_ROLE)
+    {
+        bool userMintsExceeded = mintsPerUserInCycle[msg.sender] + 1 > maxMintsPerUserInCycle;
+
+        require(msg.value >= mintFee(), "Not enough ETH!");
+
+        uint256 tokenId = currentTokenId++;
+        _safeMint(msg.sender, tokenId);
+        _setTokenURI(tokenId, uri);
+
+        if(userMintsExceeded){
+            mintsPerUserInCycle[msg.sender] = 0;
+        }
+        mintsPerUserInCycle[msg.sender]++;
+    }
+    ```
+
+    **Requirements**:  
+    - The caller **MUST** have the **CREATOR_ROLE**.  
+    - The user **MUST** pay the minting fee, which is dynamic based on their previous minting activity.  
```

Comment:
> dynamic based on their previous minting activity.

How is this calculated? Is the contract developer free to define this, or is it a particular formula? Include this information in the document, and don't just reply to me here :wink: